pkg/kubelet/types/constants.go

```diff
@@ -21,8 +21,9 @@ const (
 	ResolvConfDefault = "/etc/resolv.conf"
 
 	// different container runtimes
-	DockerContainerRuntime = "docker"
-	RemoteContainerRuntime = "remote"
+	ContainerdContainerRuntime = "containerd"
```

Comment:
It's not used. Why adding this? Also, in general, we don't want to add more container-runtime-specific things in kubelet.

Reply:
Done

Reply:
Actually, it was needed in ``pkg/kubelet/kuberuntime/kuberuntime_manager.go``. Though the code was unfortunately modified while rebasing / addressing comments and so on. We're actually looking for the ``containerd`` container runtime. It is even logged here:

    I1113 18:12:22.938271    6980 kuberuntime_manager.go:208] Container runtime containerd initialized, version: 9238d57-TEST, apiVersion: v1alpha2

You can see the ``containerRuntimeName`` used in the logging message above. That's the same thing we're looking for.

Additionally, after fixing this issue, the ``termination-log`` file is properly mounted:

        kubectl exec -n ns1 win-busybox -- powershell -command "ls C:/dev"


        Directory: C:\dev


    Mode                LastWriteTime         Length Name
    ----                -------------         ------ ----
    -a----       11/13/2019   6:25 PM              0 termination-log

Reply:
What's wrong with using "remote" or "docker" to determine?

We should not introduce more runtime strings in kubelet

Reply:
It cannot be ``remote`` since the container runtime is checked and determined. so far, for windows, it can be ``docker`` or ``containerd``. And so far, we know that single file mappings are supported only in containerd (and not in docker). If a new container runtime would be introduced in the future, there might be the possibility that it won't support support this feature, and thus, checking ``checking for ``m.Type() != types.DockerContainerRuntime`` would be incorrect. ``checking for ``m.Type() == types.ContainerdContainerRuntime`` is deterministic and certain.

Reply:
> 
> 
> > If a new container runtime would be introduced in the future, there might be the possibility that it won't support support this feature
> 

Sure, but if it doesn't, at least with this version of code it will still be able to create and start the containers, rather than always fail because it couldn't mount the ``termination-log`` file.